server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeValuesCollectorQueue.java

```diff
@@ -48,13 +50,21 @@ public int hashCode() {
         }
     }
 
+    @FunctionalInterface
+    private interface TopChangedListener {
+        void topChanged(int slot, boolean updateCompetitiveBounds) throws IOException;
```

Comment:
Do we actually need the boolean in that method? It looks like we're only ever doing something when it is true?

Reply:
The reason I added it is because we call this every time we add a slot to the queue but the first `size` times we don't want to update the bounds yet. I could change it so that we only call it when the queue is already full. But then we won't have a value for `highestCompetitiveValueGlobalOrd` until the queue fills up. I suppose that is not a problem.

Reply:
I have removed the unnecessary boolean and renamed the listener for clarity.